### ISSUE TITLE
Binding PolicyReport resource to Namespace by ownerReference

### ISSUE THREAD

**Issue by r0binak** [falcosecurity/falcosidekick#331]
**Motivation**

Using various observability tools, it is not always clear what the resource is related to / refers to

**Feature**

When generating the PolicyReport resource, I suggest adding the ownerReference field. By this field, it will be easy to determine what the resource belongs to. For example, such functionality is available in Kyverno. Very comfortably

**Comment by Issif** [comment]
Hello,

Just to be sure we're aligned, you would like something like:

```yaml
APIVersion: "v1",
Kind:       "Deployment",
Name:       <falcosidekick-namespace>,
UID:        <falcosidekick-deployment-uuid>,
Controller: false,
```

I'm not 100% sure what's the most obvious for users, using Falco or Sidekick as Owner, wdyt?

**Comment by fjogeleit** [comment]
> Hello,
> 
> Just to be sure we're aligned, you would like something like:
> 
> ```yaml
> APIVersion: "v1",
> Kind:       "Deployment",
> Name:       <falcosidekick-namespace>,
> UID:        <falcosidekick-deployment-uuid>,
> Controller: false,
> ```
> 
> I'm not 100% sure what's the most obvious for users, using Falco or Sidekick as Owner, wdyt?

Using the deployment is not possible because the owner reference can not be a namespace scoped object in another resource namespace as the report

You can bind it, as Kyverno to the Namespace object.

**Comment by Issif** [comment]
The behavior of the output is:
* if `ka.target.namespace` or  `k8s.ns.name` are present, it will create/update a Policy Report
* else it will create/update a Cluster Policy Report

I understand we can set the Namespace as Owner for the Policy Report, but what which one, falco's namespace? compromised resource's namespace?
And what about the Cluster Policy Report?

**Comment by r0binak** [comment]
> The behavior of the output is:
> 
> * if `ka.target.namespace` or  `k8s.ns.name` are present, it will create/update a Policy Report
> * else it will create/update a Cluster Policy Report
> 
> I understand we can set the Namespace as Owner for the Policy Report, but what which one, falco's namespace? compromised resource's namespace? And what about the Cluster Policy Report?

If a namespace resource, then the namespace for the resource that was attacked. Cluster not to connect in any way.

**Comment by Issif** [comment]
FYI, the ClusterPolicyReport are used for attacks to the kubernetes API from outside a Pod, these events are collected with the k8saudit log plugin

**Comment by Issif** [comment]
Does this seem good to you?

```yaml
apiVersion: wgpolicyk8s.io/v1alpha2                                                                                                                                                                                
kind: PolicyReport                                                                                                                                                                                                 
metadata:                                                                                                                                                                                                          
  creationTimestamp: "2022-07-21T15:01:08Z"                                                                                                                                                                        
  generation: 7                                                                                                                                                                                                    
  name: falco-policy-report-4dca6378                                                                                                                                                                               
  namespace: default                                                                                                                                                                                               
  ownerReferences:                                                                                                                                                                                                 
  - apiVersion: v1                                                                                                                                                                                                 
    controller: false                                                                                                                                                                                              
    kind: Namespace                                                                                                                                                                                                
    name: default                                                                                                                                                                                                  
    uid: db58b116-3b9d-45dd-b949-1624b7468fe2                                                                                                                                                                      
  resourceVersion: "169906"                                                                                                                                                                                        
  uid: f8942f93-b643-4ad7-9042-dcf87957e4af        
```             

You can test the image `falcosecurity/falcosidekick:owner`

**Comment by fjogeleit** [comment]
Is default the Namespace of the Report? I think he means the namespace where falcosidekick is installed, like kyverno sets the namespace of the kyverno pods.

**Comment by Issif** [comment]
This is not the default, the current implementation of the `PolicyReport` creates the reports in the same namespace than the Pod where the event occurred. I used for my test a Pod in `default` that's all.

**Comment by Issif** [comment]
After discussion with @fjogeleit it appeared I misunderstood the use case. I changed the code and now it adds as OwnerRef the namespace where Falcosidekick is deployed. The value is retrieved by the env var NAMESPACE that it will be set in the deployment template.

Here's the result:

```
apiVersion: wgpolicyk8s.io/v1alpha2                                                                                                                                                                                
kind: PolicyReport                                                                                                                                                                                                 
metadata:                                                                                                                                                                                                          
  creationTimestamp: "2022-07-21T20:17:17Z"                                                                                                                                                                        
  generation: 3                                                                                                                                                                                                    
  name: falco-policy-report-49560956                                                                                                                                                                               
  namespace: default                                                                                                                                                                                               
  ownerReferences:                                                                                                                                                                                                 
  - apiVersion: v1                                                                                                                                                                                                 
    controller: false                                                                                                                                                                                              
    kind: Namespace                                                                                                                                                                                                
    name: falco                                                                                                                                                                                                    
    uid: 8247029e-ba31-4de6-8748-c2bea5437498 
```

**Comment by Issif** [comment]
It will be released with 2.27.0
